packages/core/src/extensions/SideMenu/SideMenuPlugin.ts

```diff
@@ -268,30 +274,71 @@ export class SideMenuView<
     }
   };
 
-  /**
-   * If the event is outside the editor contents,
-   * we dispatch a fake event, so that we can still drop the content
-   * when dragging / dropping to the side of the editor
-   */
   onDrop = (event: DragEvent) => {
+    if (this.pmView.dragging === null) {
+      return;
+    }
+
     this.editor._tiptapEditor.commands.blur();
 
-    // ProseMirror doesn't remove the dragged content if it's dropped outside
-    // the editor (e.g. to other editors), so we need to do it manually. Since
-    // the dragged content is the same as the selected content, we can just
-    // delete the selection.
-    if (this.isDragOrigin && !this.pmView.dom.contains(event.target as Node)) {
-      this.pmView.dispatch(this.pmView.state.tr.deleteSelection());
+    if (this.sideMenuDetection === "editor") {
+      // When ProseMirror handles a drop event on the editor while
+      // `view.dragging` is set, it deletes the selected content. However, if
+      // a block from a different editor is being dropped, this causes some
+      // issues that the code below fixes:
+      //
+      // 1. Because the editor selection is unrelated to the dragged content,
+      // we don't want PM to delete its content. Therefore, we collapse the
+      // selection.
+      // 2. Because the editor from which the block originates doesn't get a
+      // drop event on it, PM doesn't delete its selected content. Therefore, we
+      // need to do so manually.
+      if (
+        !this.isDragOrigin &&
+        this.pmView.dom.contains(event.target as Node)
+      ) {
+        this.pmView.dispatch(
+          this.pmView.state.tr.setSelection(
+            TextSelection.create(
+              this.pmView.state.tr.doc,
+              this.pmView.state.tr.selection.to
+            )
+          )
+        );
+      } else if (
+        this.isDragOrigin &&
+        !this.pmView.dom.contains(event.target as Node)
+      ) {
+        // Deleting the selected content may change the height of the editor.
+        // This may cause the position of the editor in which the user is
+        // dropping to shift, before it can fire its drop event handler. The
+        // position shift can then cause the drop to happen somewhere other than
+        // the user intended.
+        //
+        // To get around this, we delay deleting the selected content until all
+        // editors have had the chance to handle the event.
+        setTimeout(
+          () => this.pmView.dispatch(this.pmView.state.tr.deleteSelection()),
+          0
+        );
+      }
+
+      // PM only clears `dragging` on the editor that the block was dropped, so
+      // we manually have to clear it on all the others.
+      setTimeout(() => (this.pmView.dragging = null), 0);
```

Comment:
why does this need to be in a settimeout?

Reply:
Since PM uses `view.dragging` when handling the drop event, we also need to delay clearing it until we're sure that's done - I'll also add a comment here